### PR TITLE
Fetches grants after wallet recovery

### PR DESCRIPTION
**Pull request by ryanml**
Fixes: https://github.com/brave/brave-browser/issues/4220

This also removes arguments that were never being used for the `RewardsService::FetchGrants` call.

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests && npm run test-security`) on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection

## Test Plan:
Defined in issue

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by NejcZdovc** [comment]
@ryanml recovery response should return grants as you can see here https://github.com/brave/brave-core/pull/2372/files#diff-2e09ca2971f3a4e74fb753ef2fca3113R133. Let's try to figure out why grants are not correct instead of adding another get

**Comment by ryanml** [comment]
@NejcZdovc those are assigned under walletInfo, which are grants already claimed. We need to get grants available for claiming, which exists at the top level of the state.

File: browser/ui/webui/brave_rewards_ui.cc [hunk 1]
```diff
@@ -472,11 +472,7 @@ void RewardsDOMHandler::OnGrant(
 
 void RewardsDOMHandler::GetGrants(const base::ListValue* args) {
   if (rewards_service_) {
-    std::string lang;
-    std::string paymentId;
-    args->GetString(0, &lang);
-    args->GetString(1, &paymentId);
-    rewards_service_->FetchGrants(lang, paymentId);
+    rewards_service_->FetchGrants(std::string(), std::string());
```
Comment (NejcZdovc) [inline browser/ui/webui/brave_rewards_ui.cc]: Is this change needed? This removes option so that JS could send lang and payment id

**Comment by NejcZdovc** [comment]
@ryanml sounds good

File: browser/ui/webui/brave_rewards_ui.cc (hunk 1 above)
Comment (ryanml) [reply browser/ui/webui/brave_rewards_ui.cc]: There is no call anywhere from the Rewards UI that pass these params (this is how it is in the panel as well) 

If you would like to keep it how it was for possible future use I am fine with that.

Comment (NejcZdovc) [reply browser/ui/webui/brave_rewards_ui.cc]: yup I know that we currently we don't use it, but if we already have it I would keep it

Comment (ryanml) [reply browser/ui/webui/brave_rewards_ui.cc]: @NejcZdovc reverted back

Comment (NejcZdovc) [reply browser/ui/webui/brave_rewards_ui.cc]: awesome thx

**Comment by NejcZdovc** [comment]
@ryanml could you please add unit test for this reducer
